data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSinkConfig.java

```diff
@@ -0,0 +1,194 @@
+package org.opensearch.dataprepper.plugins.sink.opensearch;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import lombok.Getter;
+import org.apache.commons.lang3.EnumUtils;
+import org.opensearch.dataprepper.model.opensearch.OpenSearchBulkActions;
+import org.opensearch.dataprepper.plugins.sink.opensearch.configuration.ActionConfiguration;
+import org.opensearch.dataprepper.plugins.sink.opensearch.configuration.AwsAuthenticationConfiguration;
+import org.opensearch.dataprepper.plugins.sink.opensearch.configuration.DlqConfiguration;
+import org.opensearch.dataprepper.plugins.sink.opensearch.index.TemplateType;
+
+import java.util.List;
+import java.util.Objects;
+
+public class OpenSearchSinkConfig {
+    @Getter
+    @JsonProperty("hosts")
```

Comment:
I think you need to add `JsonDescription`. Please check with Kat.

Reply:
I would suggest doing this in a follow-on PR. Leaving them out leaves us no worse off than we already are. And this is a significant change in its own right.